ReactNative/Views/LogoView.swift

```diff
@@ -0,0 +1,55 @@
+//
+// Copyright (c) 2017-2019 Cliqz GmbH. All rights reserved.
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+//
+
+import Foundation
+import React
+
+class LogoView: UIView {
+    private var url: String?
+    private lazy var reactView: UIView = {
+        guard let url = self.url else { return UIView() }
+
+        let reactView = RCTRootView(
+            bridge: ReactNativeBridge.sharedInstance.bridge,
+            moduleName: "Logo",
+            initialProperties: [
+                "url": url,
+                "size": self.bounds.width,
+            ]
+        )!
+
+        reactView.backgroundColor = .clear
+        return reactView
+    }()
+
+    override init(frame: CGRect) {
+        super.init(frame: frame)
+    }
+
+    convenience init(url: String) {
+        self.init(
+            frame: .zero,
+            url: url
+        )
+    }
+
+    convenience init(frame: CGRect, url: String) {
+        self.init(frame: frame)
+        self.url = url
+    }
+
+    required init(coder aDecoder: NSCoder) {
+        fatalError("This class does not support NSCoding")
+    }
+
+    override func layoutSubviews() {
+        super.layoutSubviews()
+        self.addSubview(self.reactView)
```

Comment:
in layoutSubviews self.addSubview(self.reactView) this is not nice. Even though the method is smart and won't add it multiple times, but it's not a good practice until you really have a good reason for this. If you simply want to add subview, do it in init of the view. layoutSubviews many many times, and here you need to only play with frames and constraints if required.

Reply:
you can move this addSubview into override init(frame: CGRect)  method